recipes/aws-c-common/all/conanfile.py

```diff
@@ -51,23 +50,21 @@ def _configure_cmake(self):
         self._cmake.configure()
         return self._cmake
 
-    def _patch_sources(self):
-        for patch in self.conan_data["patches"][self.version]:
-            tools.patch(**patch)
-
     def build(self):
-        self._patch_sources()
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
         cmake = self._configure_cmake()
         cmake.build()
 
     def package(self):
         self.copy(pattern="LICENSE", dst="licenses", src=self._source_subfolder)
         cmake = self._configure_cmake()
         cmake.install()
-
         tools.rmdir(os.path.join(self.package_folder, "lib", "aws-c-common"))
 
     def package_info(self):
+        if self.settings.os == "Linux":
+            self.cpp_info.system_libs = ["dl"]
```

Comment:
You cannot combine global properties `self.cpp.info.system_libs` with usage of components (`self.cpp_info.components`).

A few lines later in this recipe, there is already:
```python
        if self.settings.os == "Linux":
            self.cpp_info.components["aws-c-common-lib"].system_libs = ["m", "pthread", "rt"]
```
So is this addition needed?

Reply:
Ok my bad. Fixed here: c200e16